File: lib/bean/notification_config.dart
/*
* 作者 Ren
* 时间  2023/7/4 21:15
*/

import 'package:flutter/material.dart';

class NotificationConfig {
  String? appIcon; // only android
  String? androidLargeIcon; // only android
  String? androidSmallIcon; // only android
  String? androidPlayIcon; // android required
  String? androidPauseIcon; // android required
  String? androidPreIcon; // android required
  String? androidNextIcon; // android required
  String? androidPlaceholderImage; // only android
  Size? imageSize;

  NotificationConfig({
    this.appIcon,
    this.androidLargeIcon,
    this.androidSmallIcon,
    this.androidPlayIcon,
    this.androidPauseIcon,
    this.androidPreIcon,
    this.androidNextIcon,
    this.androidPlaceholderImage,
    this.imageSize,
  });

  Map toMap() {
    return {
      'appIcon': appIcon,
      'androidLargeIcon': androidLargeIcon,
      'androidSmallIcon': androidSmallIcon,
      'androidPlayIcon': androidPlayIcon,
      'androidPauseIcon': androidPauseIcon,
      'androidPreIcon': androidPreIcon,
      'androidNextIcon': androidNextIcon,
      'androidPlaceholderImage': androidPlaceholderImage,
      'imageWidth': imageSize?.width,
      'imageHeight': imageSize?.height,
    };
  }
}

File: android/src/main/kotlin/com/nl/media/notification/config/NotificationConfig.kt
package com.nl.media.notification.config

class NotificationConfig(
    var appIcon: String? = null,
    var androidLargeIcon: String? = null,
    var androidSmallIcon: String? = null,
//    var androidPlayIcon: String? = null,
//    var androidPauseIcon: String? = null,
//    var androidPreIcon: String? = null,
//    var androidNextIcon: String? = null,
    var androidPlaceholderImage: String? = null,
    var imageWidth: Double? = null,
    var imageHeight: Double? = null,
) {
    companion object {
        fun fromMap(map: Map<String, Any?>): NotificationConfig {
            return NotificationConfig(
                appIcon = map["appIcon"] as? String?,
                androidLargeIcon = map["androidLargeIcon"] as? String?,
                androidSmallIcon = map["androidSmallIcon"] as? String?,
//                androidPlayIcon = map["androidPlayIcon"] as? String?,
//                androidPauseIcon = map["androidPauseIcon"] as? String?,
//                androidPreIcon = map["androidPreIcon"] as? String?,
//                androidNextIcon = map["androidNextIcon"] as? String?,
                androidPlaceholderImage = map["androidPlaceholderImage"] as? String?,
                imageWidth = map["imageWidth"] as? Double?,
                imageHeight = map["imageHeight"] as? Double?,
            )
        }
    }
}